pkg/network/multus/status_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Authors.
+ *
+ */
+
+package multus_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8scorev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/multus"
+)
+
+var _ = Describe("Network Status", func() {
+	const multusNetworkStatusWithPrimaryNet = `[{"name":"k8s-pod-network","ips":["10.244.196.146","fd10:244::c491"],"default":true,"dns":{}}]`
+
+	DescribeTable("It should return empty", func(annotations map[string]string) {
+		result := multus.NonDefaultMultusNetworksIndexedByIfaceName(newStubPod(annotations))
+		Expect(result).To(BeEmpty())
+	},
+		Entry("when network-status is missing", map[string]string{}),
+		Entry("when network-status contains only pod network",
+			map[string]string{networkv1.NetworkStatusAnnot: multusNetworkStatusWithPrimaryNet},
+		),
+	)
+
+	It("Should return a map of pod interface name to network status when interface name is hashed", func() {
```

Comment:
this and line 76 can be a table, but on the other hand since one is old naming scheme and the other is new 
it might be fine to keep as is, wdyt ?
(personally i tend bit more to do a table, it can ease reading and make it compact)

Reply:
I don't think we should combine both tests since they produce different results - which will cause us to take the expected result as an input.

Reply:
yes of course the expected will be an input
but we do it a lot imho